receiver/signalfxreceiver/receiver.go

```diff
@@ -165,17 +171,29 @@ func (r *sfxReceiver) Shutdown(context.Context) error {
 	return err
 }
 
-func (r *sfxReceiver) readBody(ctx context.Context, resp http.ResponseWriter, req *http.Request) ([]byte, bool) {
+func (r *sfxReceiver) readContentType(ctx context.Context, resp http.ResponseWriter, req *http.Request, events bool) (bool, bool) {
 	if req.Method != http.MethodPost {
 		r.failRequest(ctx, resp, http.StatusBadRequest, invalidMethodRespBody, nil)
-		return nil, false
+		return false, false
 	}
```

Comment:
I would move this code out from this func and use is as is in `handleEventReq` and `handleDatapointReq` as it's not related to content type. Small code duplication but improves readability.